src/sensors/src/imu_subscriber.cpp

```diff
@@ -1,79 +1,73 @@
 #include "sensors/imu_subscriber.hpp"
 
-IMUSubscriber::IMUSubscriber(const std::string &csv_file_path)
-    : Node("imu_subscriber"), csv_file_path_(csv_file_path)
+IMUSubscriber::IMUSubscriber(const std::string &node_name)
+    : Node(node_name)
 {
-    // Declare and retrieve calibration offsets
+    // Declare and retrieve parameters
+    this->declare_parameter<std::string>("csv_file_path", "imu_data.csv");
     this->declare_parameter<double>("gyro_x_offset", 0.0);
     this->declare_parameter<double>("gyro_y_offset", 0.0);
     this->declare_parameter<double>("gyro_z_offset", 0.0);
     this->declare_parameter<double>("accel_x_offset", 0.0);
     this->declare_parameter<double>("accel_y_offset", 0.0);
     this->declare_parameter<double>("accel_z_offset", 0.0);
 
-    gyro_x_offset_ = this->get_parameter("gyro_x_offset").as_double();
-    gyro_y_offset_ = this->get_parameter("gyro_y_offset").as_double();
-    gyro_z_offset_ = this->get_parameter("gyro_z_offset").as_double();
-    accel_x_offset_ = this->get_parameter("accel_x_offset").as_double();
-    accel_y_offset_ = this->get_parameter("accel_y_offset").as_double();
-    accel_z_offset_ = this->get_parameter("accel_z_offset").as_double();
+    mpCsvFilePath = this->get_parameter("csv_file_path").as_string();
```

Comment:
this variable is not declared in the header file. please update the declaration in include/imu/imu_subscriber.hpp from csv_file_path to mpCsvFilePath.